internal/controller/thanosreceive_controller.go

```diff
@@ -61,7 +63,8 @@ type ThanosReceiveReconciler struct {
 	metrics  controllermetrics.ThanosReceiveMetrics
 	recorder record.EventRecorder
 
-	handler *handlers.Handler
+	handler                *handlers.Handler
+	disableConditionUpdate bool
```

Comment:
Do we really expect this is something folks would disable?

Reply:
So this isn't really an external option, it's just for easier integration testing, as condition changes cause a bunch of updates